akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala

```diff
@@ -198,7 +198,7 @@ class GraphStageLogicSpec extends StreamSpec with GraphInterpreterSpecKit {
       // note: a bit dangerous assumptions about connection and logic positions here
       // if anything around creating the logics and connections in the builder changes this may fail
       interpreter.complete(interpreter.connections(0))
-      interpreter.cancel(interpreter.connections(1))
+      interpreter.cancel(interpreter.connections(1), new RuntimeException("explicit cancellation"))
```

Comment:
Excellent idea, I'm in favour of adding such capabilities, being scarred by hours and days of guessing why something somewhere cancelled these information would have helped a lot. Will review but likely not tomorrow.